### PR TITLE
Merge 2.2 1007

### PR DESCRIPTION
**Pull request by wallyworld**
## Description of change

Merge 2.2 into develop, including:
- batch status history pruning
- separate controller and model upgrades
- api server auth issues
- fix debug log tailing

## QA steps

smoke test bootstrap

### CODE REVIEW

File: api/uniter/uniter_test.go [hunk 1]
```diff
@@ -120,6 +120,17 @@ func (s *uniterSuite) assertInScope(c *gc.C, relUnit *state.RelationUnit, inScop
 	c.Assert(ok, gc.Equals, inScope)
 }
 
+func (s *uniterSuite) patchNewState(
```
Comment (axw) [inline api/uniter/uniter_test.go]: I don't think we need this any more.

Comment (wallyworld) [inline api/uniter/uniter_test.go]: ah yes, i meant to remove that

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**